R/learner_RWeka_classif_BayesNet.R

```diff
@@ -0,0 +1,113 @@
+#' @title Classification Bayes Network Learner
+#' @author damirpolat
+#' @name mlr_learners_classif.BayesNet
+#'
+#' @description
+#' Bayes Network learning using various search algorithms.
+#' Calls [RWeka::make_Weka_classifier()] from \CRANpkg{RWeka}
+#'
+#' @section Custom mlr3 parameters:
+#' - `output_debug_info`:
+#'   - original id: output-debug-info
+#'
+#' - `do_not_check_capabilities`:
+#'   - original id: do-not-check-capabilities
+#'
+#' - `num_decimal_places`:
+#'   - original id: num-decimal-places
+#'
+#' - `batch_size`:
+#'   - original id: batch-size
+#'
+#' - Reason for change: This learner contains changed ids of the following control arguments
+#' since their ids contain irregular pattern
+#'
+#'
+#' @templateVar id classif.BayesNet
+#' @template learner
+#'
+#' @template seealso_learner
+#' @template example
+#' @export
+LearnerClassifBayesNet = R6Class("LearnerClassifBayesNet",
+  inherit = LearnerClassif,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      param_set = ps(
+        subset = p_uty(tags = "train"),
+        na.action = p_uty(tags = "train"),
+        D = p_lgl(tags = "train"),
+        B = p_uty(tags = "train"),
+        Q = p_uty(tags = "train"),
```

Comment:
I'm not really familiar with WEKA. Why are so many parameters untyped? Normally we don't tune untyped parameters.

Reply:
The tuning space includes `c("K2", "HillClimber", "LAGDHillClimber", "SimulatedAnnealing", "TabuSearch", "TAN")`.

Reply:
I turned most untyped params into factors. However, there are some that are still untyped like `W` from `AdaBoostM1` because it accepts name of a base learner. Since there are so many possible base learners to enumerate I just left it as untyped.